python/hail/expr.py

```diff
@@ -523,6 +523,18 @@ def _typecheck(self, annotation):
                                          ([f.name for f in self.fields], annotation._attrs))
                 f.typ._typecheck((annotation[f.name]))
 
+    def transitive_fields(self):
+        self._transitive_fields("", identifiers)
+
+    def _transitive_fields(self, prefix="", identifiers=[]):
+        for f in self.fields:
+            if (isinstance(f.typ, TStruct):
+                doit(f.typ, prefix+f.name+".", identifiers)
+            else:
+                identifiers.append('`{}`'.format(prefix+f.name)
+        return identifiers
```

Comment:
This would fail if you tested it twice -- you mutate the default argument of the function, which isn't reallocated each time the function is called but once per module import.

Reply:
just don't give them defaults, I think, since it's only called by transitive_fields

Reply:
er, maybe you never call with default args. Still should be fixed.

Reply:
ugh. sometimes I really can't stand python.